extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -143,6 +144,30 @@ public String name() {
         });
     }
 
+    public Supplier<Map<String, Object>> createAdditionalSecEventPropsSupplier() {
+        return new Supplier<Map<String, Object>>() {
+            @Override
+            public Map<String, Object> get() {
+                if (Arc.container().requestContext().isActive()) {
+
+                    // if present, add RoutingContext from CDI request to the SecurityEvents produced in Security extension
+                    // it's done this way as Security extension is not Vert.x based, but users find RoutingContext useful
+                    var event = Arc.container().instance(CurrentVertxRequest.class).get().getCurrent();
+                    if (event != null) {
+
+                        if (event.user() instanceof QuarkusHttpUser user) {
+                            return Map.of(RoutingContext.class.getName(), event, SecurityIdentity.class.getName(),
```

Comment:
You have good eyes, this is the weakest place of this PR and I wasn't sure how to deal with it.

- In Quarkus Security I cannot take the identity from `RoutingContext`.
- Inside `SecurityConstrainer`, I cannot call: 
https://github.com/quarkusio/quarkus/blob/main/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/SecurityIdentityAssociation.java#L62 because it would trigger authentication. I cannot add a new method like `SecurityIdentityAssociation#getIdentityIfPresent`, because visibility would have to be `public` (it's different package) and I wonder if it would confuse users [though IMHO they must use CurrentIdentityAssociation in order to program against interface].

so the identity must come from Vert.x HTTP. I put it to the properties and I absolutely agree it's strange to have the identity in properties only in this scenario. 

- The easiest way would to to simply remove it before creating the event somewhere around here (outside that IF): https://github.com/quarkusio/quarkus/blob/aeba22d7d597e5f05e9dabb4797dfebf79bc97cf/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/SecurityConstrainer.java#L132
- if that is not acceptable, I would require a new runtime SPI class, which I tried to avoid because users work with the SPI (reproducer of this issue works with AbstractSecurityEvent); plus it's one extra class for metadata that can be avoided

WDYT?

Reply:
Duplicating `SecurityIdentity` as one of the event properties is not a big problem.

What I don't understand is this: `HttpAuthorizer` already fires authorization failure events if the access was denied, and there it includes the identity, so, what case does this PR cover ?

Reply:
> Duplicating `SecurityIdentity` as one of the event properties is not a big problem.

cool, that was my thinking

> What I don't understand is this: `HttpAuthorizer` already fires authorization failure events if the access was denied, and there it includes the identity, so, what case does this PR cover ?

Unsecured JAX-RS resources that calls CDI beans with standard security annotations. Inside JAX-RS endpoint you will never be able to statically analyze what CDI beans are going to be invoked (it depends on business logic). 

If you want, you can have a look into the tests for it, it's the one called `testNestedRolesAllowed` and `testNestedPermitAll`, but I can also describe it for you. What is happening is that CDI beans are secured by CDI interceptors. and the CDI interceptors are inside Security extension that is not based on the Vert.x. In https://github.com/quarkusio/quarkus/issues/40170#issuecomment-2112133799 user described that they have requirement to log some actions for security audit and inside the issue description https://github.com/quarkusio/quarkus/issues/40170 he says _When a method (as opposed to a REST resource) is annotated with an RBAC annotation_ ..., please have a look.

Reply:
Is it because the security events are not always covered at the `HttpAuthorizer` (`quarkus-vertx-http-level`) level only and sometimes at the `quarkus-security` level ?

Reply:
> Is it because the security events are not always covered at the `HttpAuthorizer` (`quarkus-vertx-http-level`) level only and sometimes at the `quarkus-security` level ?

Yes, same as authorization. I can be even more specific:

```
@ApplicationScoped
public class SecuredBean {

  @RolesAllowed("admin")
  public String sayAdmin() {
     return "admin";
  }

}
```

In this example, the `SecuredBean#sayAdmin` is secured by CDI interceptor, not the `HttpAuthorizer`. It is not possible to use the `HttpAuthorizer` because the authorizer is for HTTP request, but this is method chain invocation. In other worlds, you do not know that `SecuredBean#sayAdmin` is going to be called.

Reply:
> Would it make sense to update `AbstractSecurityEvent.getSecurityIdentity` to check the properties if the identity property is null ? Otherwise we'd need to recommend users in the docs that you have to check both `getIdentity` and properties which may be a bit confusing for users

I didn't think of this solution. What I did here https://github.com/quarkusio/quarkus/blob/aeba22d7d597e5f05e9dabb4797dfebf79bc97cf/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/SecurityConstrainer.java#L132 is that I copied the identity from properties to the `io.quarkus.security.spi.runtime.AbstractSecurityEvent#securityIdentity`, this way, `AbstractSecurityEvent.getSecurityIdentity` works.

What you suggest would work as well. I open to changing it if you prefer that.

Reply:
What you did looks better as it can only happen on this security constrainer path